src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const mutableState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(mutableState, action.extraData);
+        break;
+
+      case 'clear':
+        for (const key in mutableState) {
+          delete mutableState[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
```

Comment:
You can use destructuring of object properties here

Reply:
Did I want to rewrite a code? Or do this just for me(Without commit)?